docs/configuration/OVERVIEW.md

```diff
@@ -156,106 +214,290 @@ export interface MyConfig extends Configuration {
    * myStringListField description
    */
   myStringListField: string[];
+
+  /**
+   * myUnionTypeField description
+   */
+  myUnionTypeField: 'before' | 'after';
+
+  /**
+   * myReferencedUnionTypeField description
+   */
+  myReferencedUnionTypeField: Position;
+
+  /**
+   * myNestedField description
+   */
+  myNestedField: MyNestedConfig[];
+}
+
+/**
+ * MyNestedConfig description
+ */
+interface MyNestedConfig extends NestedConfiguration {
+  /**
+   * myNestedStringField description
+   */
+  myNestedStringField: string;
+
+  /**
+   * myNestedBooleanField description
+   */
+  myNestedBooleanField: boolean;
+
+  /**
+   * myNestedNumberField description
+   */
+  myNestedNumberField: number;
 }
 
 export const MY_DEFAULT_CONFIG: MyConfig = {
   myStringField: 'myStringField default value',
   myBooleanField: false,
   myNumberField: 0,
-  myStringListField: ['firstDefaultValue', 'secondDefaultValue']
+  myUnionTypeField: 'before',
+  myReferencedUnionTypeField: 'top',
+  myStringListField: ['firstDefaultValue', 'secondDefaultValue'],
+  myNestedField: [
+    {
+      'myNestedStringField': 'myNestedStringField default value 1',
+      'myNestedBooleanField': false,
+      'myNestedNumberField': 15
+    },
+    {
+      'myNestedStringField': 'myNestedStringField default value 2',
+      'myNestedBooleanField': true,
+      'myNestedNumberField': 10
+    }
+  ]
 };
 ```
 
-### Component file (*.component.ts)
+> [!WARNING]
+> The default configuration const needs to be explicitly typed with the configuration interface and contain no variable references.
+
+> [!NOTE]
+> The order for Nested configuration does not matter.
+
+`UnionTypes` are supported in 2 cases:
+
+1) inline definition (see above `myUnionTypeField`)
+2) reference to a union type that is defined in the same configuration file (see above `myReferencedUnionTypeField` and `Position`).
+
+### Configuration tags
```

Comment:
Maybe we can add that we expose a linter for that `docs/linter/eslint-plugin/rules/o3r-categories-tags.md`

Reply:
And a vscode extension for autocomplete

Reply:
I will open an issue once the PR is merged

Reply:
https://github.com/AmadeusITGroup/otter/issues/1864